search/src/image_kernels.cu

```diff
@@ -0,0 +1,319 @@
+/*
```

Comment:
A lot of the content of this file is from splitting the kernels.cu file. Can this be branched off that file instead, to maintain that semantic link between them?

Reply:
I'm not seeing a good way to do that in git. There appears to be not 'git cp' type command. There are some work arounds online, but they didn't actually work for me.